Content.Server/Ghost/GhostSystem.cs

```diff
@@ -409,23 +409,41 @@ public bool DoGhostBooEvent(EntityUid target)
             return SpawnGhost(mind, spawnPosition, canReturn);
         }
 
+        private bool IsValidSpawnPosition(EntityCoordinates? spawnPosition)
+        {
+            if (spawnPosition?.IsValid(EntityManager) != true)
+                return false;
+
+            var mapUid = spawnPosition?.GetMapUid(EntityManager);
+            var gridUid = spawnPosition?.EntityId;
```

Comment:
This isn't necessarily gridUid right now but it should probably be ensured that it is in the SpawnGhost method.

I'll probably do that myself when the TryGetGridOrMapCoordinates PRs get merged